blog/_posts/2021-05-18-argocd-private-clusters.md

```diff
@@ -0,0 +1,419 @@
+---
+layout: post
+title: Learn how to manage apps across private Kubernetes clusters
+description: Learn a simple approach to deploying applications across multiple public and private Kubernetes clusters using ArgoCD, inlets PRO and GitOps.
+author: Johan Siebens
+tags: inlets-pro hybrid-cloud multi-cloud gitops argocd
+author_img: jsiebens
+image: /images/2021-05-18-argocd-private-clusters/background.jpg
+date: 2021-05-18
+---
+
+In this post, you'll learn how to apply GitOps on multiple private Kubernetes cluster with a single Argo CD installation.
+
+## Continuous Deployment on Kubernetes with GitOps
+
+[Alexis Richardson](https://twitter.com/monadic) coined the term _GitOps_ in 2017. Since then it's gained significant interest in the enterprise companies because it allows for easier auditing of deployments.
+
+With a GitOps approach, the desired state is set in several configuration files in a Git repository, and an agent component continuously compares this state with the actual state. The differences are known as drift, and the agent’s role is to detect and correct it.
+
+Two of the most popular open-source projects for GitOps are Flux, which was created at Weaveworks. Intuit, an American payroll company, created Argo CD. Both projects were donated to the Cloud Computing Foundation (CNCF) to encourage broader use and contributions.
+
+Like our [monitoring use case](https://inlets.dev/blog/2020/12/15/multi-cluster-monitoring.html), there are many reasons you may have multiple Kubernetes clusters where you want to deploy applications on with the GitOps way of working. Even more, some of the target clusters may be running in a tightly controlled private network.
+
+With Argo CD, we have two options to manage multiple private clusters. As a start, we could install Argo CD in each one of them, which could be the most straightforward approach. But it can become quite challenging to configure and maintain all those Argo CD installations when you have many clusters at hand.
+
+Argo CD cannot only apply GitOps on the cluster where it is installed, but it can also manage so-called External Clusters. With this architecture, you will have a single Argo CD installation to register all your target cluster, providing a single pane of glass for all the GitOps deployment across your clusters.
+
+The biggest challenge lies in the communication between Argo CD and the Kubernetes API services of your highly secured private clusters.
+
+![argocd](/images/2021-05-18-argocd-private-clusters/diagram.png)
```

Comment:
DOK should be DOKS, or perhaps "Kubernetes on Digital Ocean"?

Could you centralise the image too.